l1-contracts/contracts/governance/ChainAdmin.sol

```diff
@@ -38,6 +69,8 @@ contract ChainAdmin is IChainAdmin, Ownable2Step {
         require(_calls.length > 0, "No calls provided");
         // solhint-disable-next-line gas-length-in-loops
         for (uint256 i = 0; i < _calls.length; ++i) {
+            require(_validateCall(_calls[i]), "Unallowed call");
```

Comment:
I think we should all restriction checks before call execution like this:
```
1. Check restriction 1
2. Check restriction 2
...
3. Check restriction n
n+1. Execute call 1
n+2. Execute call 2
...
2*n. Execute call n
```
And not like this:

```
1. Check restriction 1
2. Execute call 1
3. Check restriction 2
4. Execute call 2
...
2*n-1. Check restriction n-1
2*n. Execute call n
```

Reply:
Another example is that in the proposed pattern it is possible to add `PermanentRollup` restriction as one of the calls and then set the DAValidatorPair as some validium stuff

Reply:
That makes sense. Initially I thought about side effects, but I see now that we should make 

```
multicall(calls1, ...)
multicall(calls2, ..)
```
to be equal to 

```
calls3 = calls1 + calls2
multicall(calls3, ...)
```